src/main.rs

```diff
@@ -151,39 +93,33 @@ pub fn process<W: io::Write>(
 /// This does not include ratchet checks, see ../README.md#ratchet-checks
 /// Instead a `ratchet::Nixpkgs` value is returned, whose `compare` method allows performing the
 /// ratchet check against another result.
-pub fn check_nixpkgs(nixpkgs_path: &Path) -> validation::Result<ratchet::Nixpkgs> {
+fn check_nixpkgs(nixpkgs_path: &Path) -> validation::Result<ratchet::Nixpkgs> {
+    let nixpkgs_path = nixpkgs_path.canonicalize()?;
```

Comment:
I think the `with_context` is important to maintain, because it's not obvious which file failed to canonicalise otherwise.

Reply:
Restored.